server/src/infra/repositories/person.repository.ts

```diff
@@ -19,64 +22,44 @@ export class PersonRepository implements IPersonRepository {
     @InjectRepository(AssetFaceEntity) private assetFaceRepository: Repository<AssetFaceEntity>,
   ) {}
 
-  /**
-   * Before reassigning faces, delete potential key violations
-   */
-  async prepareReassignFaces({ oldPersonId, newPersonId }: UpdateFacesData): Promise<string[]> {
-    const results = await this.assetFaceRepository
-      .createQueryBuilder('face')
-      .select('face."assetId"')
-      .where(`face."personId" IN (:...ids)`, { ids: [oldPersonId, newPersonId] })
-      .groupBy('face."assetId"')
-      .having('COUNT(face."personId") > 1')
-      .getRawMany();
-
-    const assetIds = results.map(({ assetId }) => assetId);
-    await this.deletePersonFromAssets(oldPersonId, assetIds);
-
-    return assetIds;
-  }
-
-  @Chunked({ paramIndex: 1 })
-  async deletePersonFromAssets(personId: string, assetIds: string[]): Promise<void> {
-    await this.assetFaceRepository.delete({ personId: personId, assetId: In(assetIds) });
-  }
-
   @GenerateSql({ params: [{ oldPersonId: DummyValue.UUID, newPersonId: DummyValue.UUID }] })
-  async reassignFaces({ oldPersonId, newPersonId }: UpdateFacesData): Promise<number> {
+  async reassignFaces({ oldPersonId, faceIds, newPersonId }: UpdateFacesData): Promise<number> {
     const result = await this.assetFaceRepository
       .createQueryBuilder()
       .update()
       .set({ personId: newPersonId })
-      .where({ personId: oldPersonId })
+      .where(
+        _.omitBy(
+          { personId: oldPersonId ? oldPersonId : undefined, id: faceIds ? In(faceIds) : undefined },
+          _.isUndefined,
+        ),
+      )
       .execute();
 
     return result.affected ?? 0;
   }
 
-  delete(entity: PersonEntity): Promise<PersonEntity | null> {
-    return this.personRepository.remove(entity);
+  async delete(entities: PersonEntity | PersonEntity[]): Promise<void> {
+    await this.personRepository.remove(entities as any);
```

Comment:
Why is this `as any` cast necessary?

Reply:
Very strangely, I get a `No overload matches this call` error if I remove the `as any`. It seems to get confused by the `PersonEntity | PersonEntity[]` typing. Doing something silly like this works:
```typescript
if (Array.isArray(entities) {
  await this.personRepository.remove(entities);
} else {
  await this.personRepository.remove(entities);
}
```

Reply:
Just cast it to one or the other then. As array or as single item. Or better yet, just make the method always use the list variant.

Reply:
Fair enough haha